src/components/file-upload/file-upload.spec.ts

```diff
@@ -83,10 +84,10 @@ describe('MFileUpload', () => {
             });
 
             const newExtensions: string[] = ['avi', 'mp3'];
-            fupd.vm.extensions = newExtensions;
+            fupd.vm.allowedExtensions = newExtensions;
 
             const newValidationOpts: MFileValidationOptions = { ...validationOpts };
-            newValidationOpts.extensions = newExtensions;
+            newValidationOpts.allowedExtensions = newExtensions;
```

Comment:
tu devrais avoir au moins 1-2 petits tests qui valide le status de tes file en fonction des allowedExtensions / rejectedExtensions.

Reply:
à moins j'me trompe de fichier

Reply:
Dans file-upload, on fait juste passer à setValidationOptions les extension acceptées et refusées. J'ai fais un test dans file.spec.ts

Reply:
de ce que je vois, dans MFileUpload, on lui passe en prop les allowedExtensions et les rejected..? si c'est le cas, en principe, les tests peuvent être très simples pas nécessairement besoin de setter plein d'extensions, seulement un et la ramplacer par un autre devrait faire la job..
Aussi, je recommenderais de diviser les tests en trois blocs clairs, // given // when // then, soit avec des commentaires et avec des sauts de ligne seulement entre ces blocs.. exemple : 
```
{code}
it('should pass validation options to $file service when extensions property is modified', async () => {
            // Given
            const filesvc: FileService = (Vue.prototype as ModulVue).$file;
            jest.spyOn(filesvc, 'setValidationOptions');

            const fupd: Wrapper<MFileUpload> = mount(MFileUpload, {
                propsData: validationOpts,
                data: {
                    isMqMinS: true
                }
            });

           // When
            const newExtensions: string[] = ['avi', 'mp3'];
            fupd.vm.allowedExtensions = newExtensions;

           // Then (mais là ici tout n'est pas vraiment un "then", donc il faudrait réarranger l'ordre, voir autre exemple)
            const newValidationOpts: MFileValidationOptions = { ...validationOpts };
            newValidationOpts.allowedExtensions = newExtensions;

            await Vue.nextTick();
            expect(filesvc.setValidationOptions).toHaveBeenCalledWith(
                newValidationOpts,
                DEFAULT_STORE_NAME
            );
        });
{code}
```

Comme je suis paresseux, je préfère simplement les espaces.. en réorganisant le code, ca donne : 
`it('should pass validation options to $file service when extensions property is modified', async () => {
            const filesvc: FileService = (Vue.prototype as ModulVue).$file;
            jest.spyOn(filesvc, 'setValidationOptions');
            const fupd: Wrapper<MFileUpload> = mount(MFileUpload, {
                propsData: validationOpts,
                data: {
                    isMqMinS: true
                }
            });
            const newExtensions: string[] = ['avi', 'mp3'];
            const newValidationOpts: MFileValidationOptions = { ...validationOpts };
            newValidationOpts.allowedExtensions = newExtensions;

            fupd.vm.allowedExtensions = newExtensions;
            await Vue.nextTick();

            expect(filesvc.setValidationOptions).toHaveBeenCalledWith(
                newValidationOpts,
                DEFAULT_STORE_NAME
            );
        });`